discord/interactions.py

```diff
@@ -213,6 +225,30 @@ def _from_data(self, data: InteractionPayload):
             except KeyError:
                 pass
 
+    @property
+    def channel(self) -> Optional[InteractionChannel]:
+        """Optional[Union[:class:`abc.GuildChannel`, :class:`abc.PrivateChannel`, :class:`Thread`]]: The channel
+        the interaction was sent from.
+
+        .. versionadded:: 2.3
+        """
+        _ch_type = self._channel.get('type')
+        if _ch_type is not None:
+            factory, ch_type = _threaded_channel_factory(_ch_type)
+            _channel_id = self._channel['id']
+            if factory is None:
+                raise InvalidData(f'Unknown channel type {_ch_type} for channel ID {_channel_id}.')
+
+            if ch_type in (ChannelType.group, ChannelType.private):
+                channel = factory(me=self._client.user, data=self._channel, state=self._state)  # type: ignore
+            else:
+                guild = self._state._get_guild(self.guild_id)
+                channel = factory(guild=guild, state=self._state, data=self._channel)  # type: ignore
+
+            return channel
+
+        return None
+
```

Comment:
None of this code belongs inside a `@property`. Removing the `_cs_channel` cached property was also a bizarre decision considering it would have made this more efficient on secondary access.

Reply:
> None of this code belongs inside a `@property`.

What about the `InvalidData` error I mentioned above?:

> Yeah, maybe better. Then, if `InvalidData` is raised, it does only effect the channel and not the whole object.

Yeah, I agree with the cached property.